architecture-evaluation/src/main/java/edu/kit/kastel/sdq/case4lang/refactorlizar/architecture_evaluation/graphs/SpoonNode.java

```diff
@@ -0,0 +1,53 @@
+package edu.kit.kastel.sdq.case4lang.refactorlizar.architecture_evaluation.graphs;
+
+import java.util.Objects;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.CtTypeMember;
+import spoon.reflect.visitor.CtVisitor;
+
+public class SpoonNode implements Node<CtType<?>> {
+    private final CtTypeMember member;
+
+    /** @param member */
+    public SpoonNode(CtTypeMember member) {
+        this.member = member;
+    }
+
+    /* (non-Javadoc)
+     * @see java.lang.Object#hashCode()
+     */
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(member, member.getDeclaringType());
+    }
+
+    /* (non-Javadoc)
+     * @see java.lang.Object#equals(java.lang.Object)
+     */
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (!(obj instanceof SpoonNode)) return false;
+        SpoonNode other = (SpoonNode) obj;
+        return Objects.equals(member, other.member)
+                && Objects.equals(member.getDeclaringType(), other.member.getDeclaringType());
+    }
```

Comment:
warning: [AlmostJavadoc] This comment contains Javadoc or HTML tags, but isn't started with a double asterisk (/**); is it meant to be Javadoc?

Reply:
This was just the comment from the original class. I did not change that.